utils/tests/test_fields.py

```diff
@@ -69,3 +73,324 @@ def test_generate_fake_data(self):
+        self.object_field.add_field(test_object_duplicate, unique=True)
+        assert test_object_duplicate not in self.object_field.object_fields
+        assert test_object in self.object_field.object_fields
+
+    def test_add_fields(self):
+        """ Tests adding multiple fields at once to the the ObjectField's list of sub-fields. Calls add_field. """
+        self.object_field.object_fields = []
+
+        field_list = [
+            BaseAPIField(name="baseField", data_type=DataType.INTEGER),
+            BaseAPIField(name="testSentence", data_type=DataType.SENTENCE),
+        ]
+        self.object_field.add_fields(field_list)
+        assert all([field in self.object_field.object_fields for field in field_list])
+
+        duplicate_field_list = [BaseAPIField(name="baseField", data_type=DataType.PHONE)]
+        self.object_field.add_fields(duplicate_field_list)  # unique=False
+        assert all([field in self.object_field.object_fields for field in duplicate_field_list])
+        assert all([field in self.object_field.object_fields for field in field_list])
+        assert len([field for field in self.object_field.object_fields if field.name == "baseField"]) == 2
+
+        unique_field = BaseAPIField(name="uniqueField", data_type=DataType.COUNTRY)
+        unique_field_list = [BaseAPIField(name="testSentence", data_type=DataType.EMAIL), unique_field]
+        self.object_field.add_fields(unique_field_list, unique=True)
+        assert len([field for field in self.object_field.object_fields if field.name == "testSentence"]) == 1
+        assert unique_field in self.object_field.object_fields
+
+    def test_combine_fields(self):
+        """
+        Tests combining fields with this ObjectField.
+        Expected behavior:
+         - If the field is a subclass of BaseAPIField, but not another ObjectField, it should simply add the field to
+           this ObjectField's list of sub-fields.
+         - If the field is another ObjectField instance, it should add all of the sub-fields of the ObjectField to this
+           ObjectField's list of sub-fields.
+        """
+        self.object_field.object_fields = []
+
+        array_field = ArrayField(name="arrayField")
+        self.object_field.combine_fields(array_field)
+        assert array_field in self.object_field.object_fields
+
+        self.object_field.combine_fields(array_field, unique=True)
+        assert len([field for field in self.object_field.object_fields if field.name == "arrayField"]) == 1
+
+        test_object = ObjectField(name="testObject")
+        test_object_subfield = BaseAPIField(name="testSentence", data_type=DataType.SENTENCE)
+        test_object.add_field(test_object_subfield)
+
+        self.object_field.combine_fields(test_object)
+        assert test_object not in self.object_field.object_fields
+        assert all([test_field in self.object_field.object_fields for test_field in test_object.object_fields])
```

Comment:
Can you explain this check?

Reply:
Yes (and I'll add a comment too):

The main purpose for the `ObjectField.combine_fields` method is to combine the fields from two distinct ObjectFields into one ObjectField. To simplify:

```
ObjectA fields:
  - name
  - date_of_birth
  
ObjectB fields:
  - favorite_color
  
ObjectA.combine_fields(ObjectB)

ObjectA fields:
  - name
  - date_of_birth
  - favorite_color
```

So this test is trying to make sure that every field that was originally in `test_object`'s list of fields is now _also_ in `self.object_field`'s list of fields. To do that, we loop through each field in `test_object.object_fields`, check if that field is also in `self.object_field.object_fields`, and then add the Boolean result of that check to a list. At the end we're checking that the list is all True - meaning every single field in `test_object` was added to `self.object_field`.

It definitely doesn't help that everything has "object" and "field" in the name 😩 . I might rename `test_object` to help with that